tensorflow/contrib/opt/python/training/nadam_optimizer_test.py

```diff
@@ -52,36 +52,43 @@ def nadam_update_numpy(param,
 class NadamOptimizerTest(test.TestCase):
 
   def doTestSparse(self, use_resource=False):
+    
+    # need to use a larger value of epsilon here so that
+    # np.sqrt(v_t) + epsilon doesn't get rounded to 0 when 
+    # the dtype is half and np.sqrt(v_t) = 0, as is the case
+    # when the gradient is 0 
+    sparse_epsilon = 1e-7
```

Comment:
I see the reasoning. Why won't it happen for dense?

Reply:
I mean since v_t is initialized to be 0, so sqrt(v_t) is 0 as well, if epsilon also gets rounded to 0 because of half float, aren't we dividing by 0 no matter dense or sparse?